### PR TITLE
ZTS: rsend tests

### PR DESCRIPTION
**Pull request by behlendorf**
### Motivation and Context

Assess the long standing `rsend` test failures which have been added
to the exceptions file to determine if they're still applicable to the latest
code.  Additionally, do a little tweaking while here to speed up the test
group.

### Description

See commits for additional details.

64f1cb22b ZTS: speed up rsend tests
ef6dda845 ZTS: rsend_007_pos failures

Note that for the `rsend_007_pos` change I haven't carefully investigated
why this avoids the issue on Linux (it is not needed for FreeBSD).  However, I
wanted to get this open for CI to have a crack at it and thought others might
have some ideas.

[Update] This minor addition does seem to reliably avoid the issue. 

### How Has This Been Tested?

Locally running the `rsend` test group on Ubuntu 20.04.  The tests
reliably pass in my testing, this PR has been opened to verify that
is the case on all other test platforms.

```
$ ./scripts/zfs-tests.sh -T rsend
Test: /home/ubuntu/zfs/tests/zfs-tests/tests/functional/rsend/setup (run as root) [00:02] [PASS]
Test: /home/ubuntu/zfs/tests/zfs-tests/tests/functional/rsend/recv_dedup (run as root) [00:00] [PASS]
Test: /home/ubuntu/zfs/tests/zfs-tests/tests/functional/rsend/recv_dedup_encrypted_zvol (run as root) [00:06] [PASS]
Test: /home/ubuntu/zfs/tests/zfs-tests/tests/functional/rsend/rsend_001_pos (run as root) [00:04] [PASS]
Test: /home/ubuntu/zfs/tests/zfs-tests/tests/functional/rsend/rsend_002_pos (run as root) [00:03] [PASS]
Test: /home/ubuntu/zfs/tests/zfs-tests/tests/functional/rsend/rsend_003_pos (run as root) [00:03] [PASS]
Test: /home/ubuntu/zfs/tests/zfs-tests/tests/functional/rsend/rsend_004_pos (run as root) [00:06] [PASS]
Test: /home/ubuntu/zfs/tests/zfs-tests/tests/functional/rsend/rsend_005_pos (run as root) [00:08] [PASS]
Test: /home/ubuntu/zfs/tests/zfs-tests/tests/functional/rsend/rsend_006_pos (run as root) [00:05] [PASS]
Test: /home/ubuntu/zfs/tests/zfs-tests/tests/functional/rsend/rsend_007_pos (run as root) [00:08] [PASS]
Test: /home/ubuntu/zfs/tests/zfs-tests/tests/functional/rsend/rsend_008_pos (run as root) [00:00] [SKIP]
Test: /home/ubuntu/zfs/tests/zfs-tests/tests/functional/rsend/rsend_009_pos (run as root) [00:03] [PASS]
Test: /home/ubuntu/zfs/tests/zfs-tests/tests/functional/rsend/rsend_010_pos (run as root) [00:00] [PASS]
Test: /home/ubuntu/zfs/tests/zfs-tests/tests/functional/rsend/rsend_011_pos (run as root) [00:05] [PASS]
Test: /home/ubuntu/zfs/tests/zfs-tests/tests/functional/rsend/rsend_012_pos (run as root) [00:14] [PASS]
Test: /home/ubuntu/zfs/tests/zfs-tests/tests/functional/rsend/rsend_013_pos (run as root) [00:08] [PASS]
Test: /home/ubuntu/zfs/tests/zfs-tests/tests/functional/rsend/rsend_014_pos (run as root) [00:05] [PASS]
Test: /home/ubuntu/zfs/tests/zfs-tests/tests/functional/rsend/rsend_016_neg (run as root) [00:00] [PASS]
Test: /home/ubuntu/zfs/tests/zfs-tests/tests/functional/rsend/rsend_019_pos (run as root) [00:29] [PASS]
Test: /home/ubuntu/zfs/tests/zfs-tests/tests/functional/rsend/rsend_020_pos (run as root) [00:12] [PASS]
Test: /home/ubuntu/zfs/tests/zfs-tests/tests/functional/rsend/rsend_021_pos (run as root) [00:14] [PASS]
Test: /home/ubuntu/zfs/tests/zfs-tests/tests/functional/rsend/rsend_022_pos (run as root) [00:13] [PASS]
Test: /home/ubuntu/zfs/tests/zfs-tests/tests/functional/rsend/rsend_024_pos (run as root) [00:13] [PASS]
Test: /home/ubuntu/zfs/tests/zfs-tests/tests/functional/rsend/send-c_verify_ratio (run as root) [01:01] [PASS]
Test: /home/ubuntu/zfs/tests/zfs-tests/tests/functional/rsend/send-c_verify_contents (run as root) [00:04] [PASS]
Test: /home/ubuntu/zfs/tests/zfs-tests/tests/functional/rsend/send-c_props (run as root) [00:20] [PASS]
Test: /home/ubuntu/zfs/tests/zfs-tests/tests/functional/rsend/send-c_incremental (run as root) [00:11] [PASS]
Test: /home/ubuntu/zfs/tests/zfs-tests/tests/functional/rsend/send-c_volume (run as root) [00:03] [PASS]
Test: /home/ubuntu/zfs/tests/zfs-tests/tests/functional/rsend/send-c_zstreamdump (run as root) [00:02] [PASS]
Test: /home/ubuntu/zfs/tests/zfs-tests/tests/functional/rsend/send-c_lz4_disabled (run as root) [00:13] [PASS]
Test: /home/ubuntu/zfs/tests/zfs-tests/tests/functional/rsend/send-c_recv_lz4_disabled (run as root) [00:16] [PASS]
Test: /home/ubuntu/zfs/tests/zfs-tests/tests/functional/rsend/send-c_mixed_compression (run as root) [00:13] [PASS]
Test: /home/ubuntu/zfs/tests/zfs-tests/tests/functional/rsend/send-c_stream_size_estimate (run as root) [00:08] [PASS]
Test: /home/ubuntu/zfs/tests/zfs-tests/tests/functional/rsend/send-c_embedded_blocks (run as root) [00:30] [PASS]
Test: /home/ubuntu/zfs/tests/zfs-tests/tests/functional/rsend/send-c_resume (run as root) [00:14] [PASS]
Test: /home/ubuntu/zfs/tests/zfs-tests/tests/functional/rsend/send-cpL_varied_recsize (run as root) [00:22] [PASS]
Test: /home/ubuntu/zfs/tests/zfs-tests/tests/functional/rsend/send-c_recv_dedup (run as root) [00:00] [PASS]
Test: /home/ubuntu/zfs/tests/zfs-tests/tests/functional/rsend/send-L_toggle (run as root) [00:01] [PASS]
Test: /home/ubuntu/zfs/tests/zfs-tests/tests/functional/rsend/send_encrypted_hierarchy (run as root) [00:05] [PASS]
Test: /home/ubuntu/zfs/tests/zfs-tests/tests/functional/rsend/send_encrypted_props (run as root) [00:05] [PASS]
Test: /home/ubuntu/zfs/tests/zfs-tests/tests/functional/rsend/send_encrypted_truncated_files (run as root) [00:01] [PASS]
Test: /home/ubuntu/zfs/tests/zfs-tests/tests/functional/rsend/send_freeobjects (run as root) [00:00] [PASS]
Test: /home/ubuntu/zfs/tests/zfs-tests/tests/functional/rsend/send_realloc_files (run as root) [01:07] [PASS]
Test: /home/ubuntu/zfs/tests/zfs-tests/tests/functional/rsend/send_realloc_encrypted_files (run as root) [01:07] [PASS]
Test: /home/ubuntu/zfs/tests/zfs-tests/tests/functional/rsend/send_spill_block (run as root) [00:53] [PASS]
Test: /home/ubuntu/zfs/tests/zfs-tests/tests/functional/rsend/send_holds (run as root) [00:02] [PASS]
Test: /home/ubuntu/zfs/tests/zfs-tests/tests/functional/rsend/send_hole_birth (run as root) [00:14] [PASS]
Test: /home/ubuntu/zfs/tests/zfs-tests/tests/functional/rsend/send_mixed_raw (run as root) [00:03] [PASS]
Test: /home/ubuntu/zfs/tests/zfs-tests/tests/functional/rsend/send-wR_encrypted_zvol (run as root) [00:03] [PASS]
Test: /home/ubuntu/zfs/tests/zfs-tests/tests/functional/rsend/send_partial_dataset (run as root) [00:02] [PASS]
Test: /home/ubuntu/zfs/tests/zfs-tests/tests/functional/rsend/send_invalid (run as root) [00:00] [PASS]
Test: /home/ubuntu/zfs/tests/zfs-tests/tests/functional/rsend/send_doall (run as root) [00:00] [PASS]
Test: /home/ubuntu/zfs/tests/zfs-tests/tests/functional/rsend/cleanup (run as root) [00:00] [PASS]
Test (Linux): /home/ubuntu/zfs/tests/zfs-tests/tests/functional/rsend/setup (run as root) [00:02] [PASS]
Test (Linux): /home/ubuntu/zfs/tests/zfs-tests/tests/functional/rsend/send_realloc_dnode_size (run as root) [00:07] [PASS]
Test (Linux): /home/ubuntu/zfs/tests/zfs-tests/tests/functional/rsend/send_encrypted_files (run as root) [00:21] [PASS]
Test (Linux): /home/ubuntu/zfs/tests/zfs-tests/tests/functional/rsend/cleanup (run as root) [00:00] [PASS]

Results Summary
PASS      56
SKIP       1

Running Time:   00:10:56
Percent passed: 98.2%
Log directory:  /var/tmp/test_results/20211217T234122

Tests with results other than PASS that are expected:
    SKIP rsend/rsend_008_pos (https://github.com/openzfs/zfs/issues/6066)

Tests with result of PASS that are unexpected:

Tests with results other than PASS that are unexpected:
```

### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Performance enhancement (non-breaking change which improves efficiency)
- [x] Code cleanup (non-breaking change which makes code smaller or more readable)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Library ABI change (libzfs, libzfs\_core, libnvpair, libuutil and libzfsbootenv)
- [x] Documentation (a change to man pages or other documentation)

### Checklist:

- [x] My code follows the OpenZFS [code style requirements](https://github.com/openzfs/zfs/blob/master/.github/CONTRIBUTING.md#coding-conventions).
- [x] I have updated the documentation accordingly.
- [x] I have read the [**contributing** document](https://github.com/openzfs/zfs/blob/master/.github/CONTRIBUTING.md).
- [ ] I have added [tests](https://github.com/openzfs/zfs/tree/master/tests) to cover my changes.
- [x] I have run the ZFS Test Suite with this change applied.
- [x] All commit messages are properly formatted and contain [`Signed-off-by`](https://github.com/openzfs/zfs/blob/master/.github/CONTRIBUTING.md#signed-off-by).

### CODE REVIEW

File: tests/zfs-tests/tests/functional/rsend/rsend.kshlib [hunk 1]
```diff
@@ -121,6 +121,7 @@ function cleanup_pool
 	log_must rm -rf $BACKDIR/*
 
 	if is_global_zone ; then
+		log_must df >/dev/null
```
Comment (freqlabs) [inline tests/zfs-tests/tests/functional/rsend/rsend.kshlib:124]: Is this some magic?

**Review by freqlabs — APPROVED** [review]
Nice to get those off the list!

File: tests/zfs-tests/tests/functional/rsend/rsend.kshlib (hunk 1 above)
Comment (behlendorf) [reply tests/zfs-tests/tests/functional/rsend/rsend.kshlib:124]: It's terrible, terrible black magic.  I'm not sure why it avoids the issue, but it does.

**Review by tonynguien — APPROVED** [review]

File: tests/zfs-tests/tests/functional/rsend/rsend.kshlib (hunk 1 above)
Comment (behlendorf) [reply tests/zfs-tests/tests/functional/rsend/rsend.kshlib:124]: See #12898 for the reason this works.  I've used that PR to add a comment here which references the root issue, at least until it's properly addressed.

**Pull request closed without merging**